### PR TITLE
chore: configure agent log level through config file

### PR DESCRIPTION
**Pull request by amandavialva01**
## Description

Agent log level can now be set via agent config.

## Test Plan

- Change the value of `log.level` (and `log.color`) in the `agent` stage of `.../determined/tools/devcluster.yaml`:
```
- agent:
    ...
    config_file:
        log:
            level: <input_level>
            color: <input_color>    
```

- Run `make devcluster` and verify that the output of `agent configuration` in the build logs includes `"log":{"level":<input_level>, "color":<input_color>}`. 

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
DET-8884

### CODE REVIEW

File: agent/internal/core.go [hunk 1]
```diff
@@ -29,6 +30,9 @@ func Run(parent context.Context, version string, opts options.Options) error {
 	wg := errgroupx.WithContext(ctx)
 
 	log.Trace("starting main agent process")
+
+	logger.SetLogrus(opts.Log)
```
Comment (corban-beaird) [inline agent/internal/core.go:34]: What happens if a log level isn't specified in the devcluster? Do we default to a specific level?

File: tools/devcluster.yaml [hunk 2]
```diff
@@ -70,3 +70,5 @@ stages:
         master_host: 127.0.0.1
         master_port: 8080
         container_master_host: $DOCKER_LOCALHOST
+        log:
+          level: trace
```
Comment (corban-beaird) [inline tools/devcluster.yaml]: Should a release note be added to communicate to end users that this config is now available?

File: agent/internal/core.go (hunk 1 above)
Comment (amandavialva01) [reply agent/internal/core.go:34]: Yea, default log level is specified [here](https://github.com/determined-ai/determined/blob/main/agent/internal/options/options.go#L28)

File: tools/devcluster.yaml (hunk 2 above)
Comment (amandavialva01) [reply tools/devcluster.yaml]: Added!

**Review by corban-beaird — APPROVED** [review]
Looks solid!

**Pull request merged**